src/lib_context/context_impl.ml

```diff
@@ -110,31 +110,17 @@ let reporter () =
   in
   {Logs.report}
 
-let index_log_size = ref None
+let index_log_size = Env.(v.index_log_size)
```

Comment:
This commit (d51fbedbf9d5e2beb100ab1b84069b82deaa4b48) should be upstreamed separately (it's a good change on its own).

Reply:
(also for only accessing fields, I would write it as `v.Env.index_log_size` and so on)

Reply:
Yes, I can get rid of the global variable and replace it with a `Env.get : unit -> Env.t` thunk that derives `Sys.argv` each time it is called.